nncf/quantization/algorithms/accuracy_control/ranker.py

```diff
@@ -154,24 +161,36 @@ def rank_groups_of_quantizers(
         :param initial_model: Initial not quantized model.
         :param quantized_model: Quantized model.
         :param quantized_model_graph: NNCF graph for quantized model.
+        :param reference_values_for_each_item: List of reference values.
+        :param approximate_values_for_each_item: List of approximate values.
         :return: List of ranked groups of quantizers.
         """
-        # See `Ranker.__init__()` to understand why we should do this.
-        if self._ref_values is None:
-            nncf_logger.info("Collecting metrics for each data item using an initial model")
-            with timer():
-                self._ref_values = self._collect_values_for_each_item(initial_model, self._get_data_items())
-
-        nncf_logger.info("Collecting metrics for each data item using a quantized model")
-        with timer():
-            approx_values = self._collect_values_for_each_item(quantized_model, self._get_data_items())
+        if self._ranking_fn is None:
+            self._ranking_fn = self._create_ranking_fn(get_backend(initial_model))
+
+        if reference_values_for_each_item is None:
+            if self._reference_values_for_each_item is None:
+                nncf_logger.info("Collecting metrics for each data item using an initial model")
+                with timer():
+                    self._reference_values_for_each_item = self._evaluator.collect_values_for_each_item(
+                        initial_model, self._dataset
+                    )
+        else:
+            self._reference_values_for_each_item = reference_values_for_each_item
+
+        if approximate_values_for_each_item is None:
+            nncf_logger.info("Collecting metrics for each data item using a quantized model")
+            approximate_values_for_each_item = self._evaluator.collect_values_for_each_item(
+                quantized_model, self._dataset
+            )
 
         # Create a subset of data items that will be used to rank groups of quantizers.
-        scores = [self._ranking_fn(ref_val, approx_val) for ref_val, approx_val in zip(self._ref_values, approx_values)]
+        scores = [
+            self._ranking_fn(ref_val, approx_val)
+            for ref_val, approx_val in zip(self._reference_values_for_each_item, approximate_values_for_each_item)
```

Comment:
Since we use `zip` almost everywhere for `reference` and `approximate` values, can we be sure that these values match each other while every `zip`?

Reply:
Yes, we can. We use the same dataset to collect references values and approximate values.